frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -119,9 +119,16 @@ export const XYChart = ({
 
   const defaultYScale = yScaleLeft || yScaleRight;
 
-  const { leftColumn, rightColumn } = getLegendColumns(series);
+  const legendItems = calculateLegendItems(
+    series.map(series => ({ name: series.name, color: series.color })),
+    width - CHART_PADDING * 2,
+    style.legend.lineHeight,
+    style.legend.fontSize,
+    style.legend.fontWeight,
+  );
+  const legendItemsHeight = legendItems.length * style.legend.lineHeight;
```

Comment:
Thanks, I fixed that. It is awkward to return an array of rows for no reason since we already calculate vertical offsets for items and we render just a plain list of rows